tomcat/src/main/java/org/apache/coyote/controller/LoginController.java

```diff
@@ -0,0 +1,77 @@
+package org.apache.coyote.controller;
+
+import java.util.Optional;
+import nextstep.jwp.db.InMemoryUserRepository;
+import nextstep.jwp.model.User;
+import org.apache.coyote.domain.request.HttpRequest;
+import org.apache.coyote.domain.response.HttpResponse;
+import org.apache.coyote.domain.response.RedirectUrl;
+import org.apache.coyote.domain.response.ResponseBody;
+import org.apache.coyote.domain.response.statusline.ContentType;
+import org.apache.coyote.domain.response.statusline.HttpStatusCode;
+import org.apache.coyote.session.Session;
+import org.apache.coyote.session.SessionManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LoginController extends AbstractController {
+
+    private static final Logger log = LoggerFactory.getLogger(LoginController.class);
+
+    private static final String URL = "/login";
+    private static final String INDEX_HTML = "/index.html";
+    private static final String USER_SESSION_KEY = "user";
+    private static final String HTML_401_PAGE = "/401.html";
+    private static final String PASSWORD_KEY = "password";
+    private static final String ACCOUNT_KEY = "account";
+
+    @Override
+    void doGet(HttpRequest request, HttpResponse response) throws Exception {
+        if (request.hasJSESSIONID() && request.checkSession()) {
+            response.responseLine(request.getHttpVersion(), HttpStatusCode.FOUND)
+                    .header(RedirectUrl.from(INDEX_HTML));
+            log.info("[Login Controller] doGet - Exist Session ");
+            return;
+        }
+        response.responseLine(request.getHttpVersion(), HttpStatusCode.OK)
+                .header(ContentType.from(request.getFilePath()))
+                .responseBody(ResponseBody.from(request.getFilePath()));
+        log.info("[Login Controller] doGet - Not Exist Session ");
+    }
+
+    @Override
+    void doPost(HttpRequest request, HttpResponse response) throws Exception {
+        Optional<User> user = InMemoryUserRepository.findByAccount(request.getBodyValue(ACCOUNT_KEY));
+        if (user.isPresent()) {
+            log.info(user.get().toString());
```

Comment:
```suggestion
            log.info("user={}", user.get());
```
해당 경우처럼 user.get().toString()일 경우 log.info 메서드 보다 먼저 실행되기 때문에 log.info보다 낮은 레벨로 loggin 레벨을 정할 시 toString()이 먼저 실행되어서 낭비가 발생할겁니다.

Reply:
메서드 호출 순서에 대해서는 신경쓰지 못했는데 또 하나  배워가네요!! 감사합니다.